examples/interactive_bluefog_helloworld.ipynb

```diff
@@ -0,0 +1,634 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "engine": 2,
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "%%px  --targets 1:3\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "plt.semilogy(mse)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# What happens after you run `ibfrun`\n",
+    "\n",
+    "If you are curious about what happened after running the ibfrun, the following figure should give you a good overview.\n",
+    "\n",
+    "<img src=\"./images/ibfrun.png\" alt=\"ibfrun\" width=\"780\"/>\n",
+    "\n",
+    "When you start the jupyter notebook, you will create a IPython client, which will keep recieving and executing the command from the notebook. `ibfrun` will create a controller and bunch of workers (it is just IPython Kernel actually) under the MPI environment. The controller knows how to communicate with the IPython client and the workers. When you execute the code under `px` environment, what happens:\n",
+    "\n",
+    "1. Send code back to the IPython client\n",
+    "2. The IPython client notices this code is need to be run under `px` so it forward the code to the controller.\n",
+    "3. The controller forward the code to the worker according the targets. By default, it will forward to all workers.\n",
+    "4. The workers accept the command and execute it in parallel. If there is data exchange between the processes, it will communicate through the MPI.\n",
+    "5. When all workers finished the executing the code, it will reverse the path to tell the jupyter noteboo that execute is done.\n",
```

Comment:
After all workers finish executing the code, the jupyter notebook will be notified that the execution is done.